suite-native/receive/src/screens/ReceiveModalScreen.tsx

```diff
@@ -79,12 +106,29 @@ const ReceiveModalScreenSubHeader = ({ accountKey, tokenContract }: ScreenSubHea
 
 export const ReceiveModalScreen = () => {
     const {
-        params: { accountKey, tokenContract },
+        params: {
+            accountKey: routeAccountKey,
+            tokenContract,
+            networkSymbol: routeNetworkSymbol,
+            accountType: routeAccountType,
+            accountIndex: routeAccountIndex,
+        },
     } = useRoute<RouteProp<RootStackParamList, RootStackRoutes.ReceiveModal>>();
 
+    const foundAccountKey = useSelector((state: AccountsRootState & DeviceRootState) =>
+        selectDeviceAccountsForNetworkSymbolAndAccountTypeWithIndex(
+            state,
+            routeNetworkSymbol,
+            routeAccountType,
+            routeAccountIndex,
+        ),
+    )?.key;
+
     const navigation =
         useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.ReceiveModal>>();
 
+    const accountKey = routeAccountKey ?? foundAccountKey;
```

Comment:
Why doesn't the selector return `foundAccountKey` as default?

Reply:
You're doing the same thing in `AccountDetailScreen`

Reply:
Just move the .key as discussed

Reply:
ab1ff66d66